### ISSUE TITLE
Function names and intent

### ISSUE THREAD

**Issue by Frikki** [dekujs/deku#131]
I have a bit of an issue with the `setState` function. Usually, such a name implies that it is a property mutator. However, that, at least not its pure form, seems to be the case. Here is why.

```javascript
function initialState() {
  return { a: 5, b: 10 };
}
```

Inspecting the `component` object confirms that the component’s `state` property is indeed `{ a: 5, b: 10 }`. That’s all good and expected.

Now, in a event handler, let’s try to set the `state` property by using the provided `setState` function.

```javascript
function someEventHandler( event, component, setState ) {
  setState( { a: 2, c: 7 } );
}
```

And now inspect the `state` property. If `setState` were indeed a mutator, we’d expect `state = { a: 2, c: 7 }`, but that is not the case. Instead we get `{ a: 2, b: 10, c: 7 }`. `setState` actually "merges" the object   into the previous `state`. However, this is _only_ a shallow merge/copy/assign. If the `state` has nested objects as properties, they won’t get merged but simply replaced ( as in the example with `a: 5` to `a: 2` ). Thus, the name `setState` is kind of misleading, as function names should communicate intent, preferably unequivocally.

**Comment by nathggns** [comment]
`setState` does the same thing in React. Maybe call it mergeState if you have an issue?

**Comment by Frikki** [comment]
Well, React is no authority on good naming. Yes, I decided to call it `updateState`, which, btw., goes well in accordance with deku’s internal `updateEntityState` behind the scenes.

**Comment by anthonyshort** [comment]
That's actually a really good point. `set` implies you're replacing it. It might be a good idea to update the docs to use `updateState` instead.

I've been playing with the idea that the handlers/hooks could return state (or a promise that resolves into state) in #54, so we wouldn't even need the method and they could remain pure.
